### PR TITLE
Us1762585 - iOS Refactor Native components

### PR DESCRIPTION
**Pull request by abaeza-wp**
## What
- Refactor ios Bridge and demo application

## Why
- Avoids having to handle inputs in states by using native components and an input field id in order to configure validation and session generation for a field

## How
- Uses the iOS bridge
- Removes hanling of input with states
- Only a input field id is needed in order to configure validation and session generation for a field
- Integrate the latest AccessCheckoutUITextField changes
- Updates and renames cvcValue to cvcId
- Updates and renames panValue to panId
- Updates and renames expiryDateValue to expiryDateId

### CODE REVIEW

File: access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/AccessCheckoutReactNativeCardValidationAcceptanceTests.swift [hunk 1]
```diff
@@ -132,10 +132,11 @@ class AccessCheckoutReactNativeCardValidationAcceptanceTests: XCTestCase {
         let accessCheckoutReactNative = AccessCheckoutReactNativeTestImplementation(
             reactNativeViewLocatorMock!)
 
-        accessCheckoutReactNative.initialiseCardValidation(config: config) { (success) in
+        accessCheckoutReactNative.initialiseCardValidation(config: config) { success in
             XCTAssertEqual(true, (success as! Bool))
 
-            self.panUITextField!.insertText("4444333322221111")
+            let field = self.panUITextField!.viewWithTag(AccessCheckoutUIFieldIdentifier.UITextField) as! UITextField
+            field.insertText("4444333322221111")
```
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/AccessCheckoutReactNativeCardValidationAcceptanceTests.swift:140]: Remove viewWithTag and replace with @testable approach

File: access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/AccessCheckoutReactNativeSessionAcceptanceTests.swift [hunk 2]
```diff
@@ -5,8 +5,35 @@ import XCTest
...
+    override func setUp() {
+        controller =
+            (storyboard.instantiateViewController(
+                withIdentifier: "SessionGenerationTestUIViewController")
+                as! SessionGenerationTestUIViewController)
+        controller!.loadViewIfNeeded()
+
+        panUITextField = controller!.panTextField
+        expiryDateUITextField = controller!.expiryDateTextField
+        cvcUITextField = controller!.cvcTextField
```
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/AccessCheckoutReactNativeSessionAcceptanceTests.swift:28]: rename UITextFields to AccessCheckoutUITextField

File: access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/AccessCheckoutReactNativeSessionAcceptanceTests.swift [hunk 3]
```diff
@@ -5,8 +5,35 @@ import XCTest
...
+
+        reactNativeViewLocatorMock = ReactNativeViewLocatorMock(
+            panUITextField: panUITextField!,
+            expiryDateUITextField: expiryDateUITextField!,
+            cvcUITextField: cvcUITextField!)
+
```
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/AccessCheckoutReactNativeSessionAcceptanceTests.swift:34]: same as Above
```suggestion
            panView: panUITextField!,
            expiryDateView: expiryDateUITextField!,
            cvcView: cvcUITextField!)

```

File: access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/session/GenerateSessionConfigUnitTests.swift [hunk 4]
```diff
@@ -76,9 +76,9 @@ class GenerateSessionUnitTests: XCTestCase {
 
     // MARK: Testing errors related to Pan Value
```
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/ios/AccessCheckoutReactNativeUnitTests/session/GenerateSessionConfigUnitTests.swift:77]: Update coments

File: access-checkout-react-native-sdk/ios/scripts/coverage.sh
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/ios/scripts/coverage.sh:1]: revert file permissions

File: access-checkout-react-native-sdk/lib/commonjs/ui/AccessCheckoutEditText.js
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/lib/commonjs/ui/AccessCheckoutEditText.js:1]: Rename to AccessCheckoutTextInput

File: access-checkout-react-native-sdk/lib/typescript/index.d.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import AccessCheckoutReactNative from './AccessCheckoutReactNative';
 export default AccessCheckoutReactNative;
 export { default as AccessCheckout } from './AccessCheckout';
-export { default as CardDetails } from './session/CardDetails';
+export { default as CardDetails } from './session/SessionGenerationConfig';
```
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/lib/typescript/index.d.ts:4]: Rename anything with CardDetails appropriately

File: access-checkout-react-native-sdk/src/session/SessionGenerationConfig.tsx
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/src/session/SessionGenerationConfig.tsx:1]: rename file extension to ts

File: access-checkout-react-native-sdk/src/ui/AccessCheckoutEditText.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Composes `AccessCheckoutEditText`.
+ *
+ * - nativeID: string
+ * - testID: string
+ * - style: StyleProp
+ * - editable: boolean
+ * - placeholder: string
+ */
+interface AccessCheckoutEditTextProps extends ViewProps {
```
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/src/ui/AccessCheckoutEditText.tsx:18]:
```suggestion
interface AccessCheckoutTextInputProps extends ViewProps {
```

File: access-checkout-react-native-sdk/src/ui/AccessCheckoutEditText.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import {
+  requireNativeComponent,
+  StyleProp,
+  TextStyle,
+  type ViewProps,
+} from 'react-native';
+
+/**
+ * Composes `AccessCheckoutEditText`.
```
Comment (abaeza-wp) [inline access-checkout-react-native-sdk/src/ui/AccessCheckoutEditText.tsx:10]:
```suggestion
 * Composes `AccessCheckoutTextInput`.
```

File: demo-app/ios/Podfile [hunk 8]
```diff
@@ -30,6 +30,8 @@ target 'AccessCheckoutReactNativeDemo' do
 
   # Remove once version 3.0.0 of the AccessCheckoutSDK has been released to Cocoapods
   pod 'AccessCheckoutSDK', :git => 'git@github.com:Worldpay/access-checkout-ios.git', :tag => 'v3.0.0'
+
+
```
Comment (abaeza-wp) [inline demo-app/ios/Podfile:34]:
```suggestion
```

File: demo-app/src/cvc-flow/CvcFlow.tsx [hunk 9]
```diff
@@ -74,12 +73,14 @@ export default function CvcFlow() {
     setShowSpinner(true);
     setIsEditable(false);
 
-    const cardDetails: CardDetails = {
-      cvc: cvcValue,
+    const sessionGenerationConfig: SessionGenerationConfig = {
+      panId: 'panInput',
+      expiryDateId: 'expiryDateInput',
```
Comment (abaeza-wp) [inline demo-app/src/cvc-flow/CvcFlow.tsx:78]:
```suggestion
```

**Review by ochalet-wp — APPROVED** [review]

**Review by ochalet-wp — APPROVED** [review]

**Pull request merged**